Automattic/jetpack
# Social: Update SIG init logic and update plan checks to check if SIG is enabled

## Proposed changes:

* Moved SIG init to admin_init
* Updated the `is_social_image_generator_enabled` function in `class-publicize-base` to use `Current_Plan`
* Added some code in `class-publicize-setup` to refresh plan data on every `jetpack_heartbeat`

### Other information:


## Jetpack product discussion


pdrWKz-P7-p2

## Does this pull request change what data or activity we track or use?


No
## Testing instructions:






* Make sure no blog stickers are added to the site
* Purchase the advanced plan on your Jetpack site and add the blog sticker for SIG. 
* On your localhost, run `jetpack docker wp shell` and inside the shell, execute `do_action('jetpack_heartbeat');`. 
* Go to your publicize sidebar on the editor and see that SIG is enabled. 
* Remove the blog sticker 
* On your localhost, run `jetpack docker wp shell` and inside the shell, execute `do_action('jetpack_heartbeat');`. 
* Go to your publicize sidebar on the editor and see that SIG is not enabled.
